src/Compiler/TaskManager.cpp

```diff
@@ -127,16 +127,18 @@ void TaskManager::startAll() {
   m_runStack.append(m_tasks[TIMING_SIGN_OFF]);
   m_runStack.append(m_tasks[POWER]);
   m_runStack.append(m_tasks[BITSTREAM]);
-  run();
+  m_taskCount = m_runStack.count();
   emit started();
+  run();
```

Comment:
Why was this moved after "emit started()" ?

Reply:
because run() will not exit until task is running

Reply:
I didn't follow why run() was before emit started() before this PR. It seems like we are changing a behavior which I am trying to understand.

Reply:
Before this change there was nothing for started() signal. Nothing was connected. No any functionality change

Reply:
Thanks, I see the connection now. 
https://github.com/os-fpga/FOEDAG/pull/554/files#diff-9e31125b4e2b6bd449e09a6b478450cb54bf4fb9ddb23baadb06ad5d9a9bb149R402